llama.cpp

```diff
@@ -696,6 +698,9 @@ struct llama_model_loader {
 
         size_t done_size = 0;
         for (llama_load_tensor & lt : tensors_map.tensors) {
+            if (lt.ggml_tensor->backend != GGML_BACKEND_CPU) {
+                continue;
+            }
```

Comment:
This looks like it may break the `progress_callback`.

Reply:
The CPU layers are at the front and the GPU layers are at the back. So the current behavior is that progress will jump from some percentage to 100% and then the GPU layers will load without calling the progress callback. Again, the problem is that this is llama functionality that presumably should not be in `ggml-cuda.cu`.

Reply:
If I am not missing something, `done_size` is not increased and `data_size` doesn't seem to be updated to remove the GPU layers either, so the progress will never reach 100%.

Reply:
After the loop in line 716 it's always set to 100%.